loadbalancer.go

```diff
@@ -28,12 +30,22 @@ func (lb RandomLoadBalancer[T]) Name() LoadBalancerPolicy {
 }
 
 // Resolve return the resolved option for Random LB
-func (lb RandomLoadBalancer[T]) Resolve(dbs []T) T {
+func (lb *RandomLoadBalancer[T]) Resolve(dbs []T) T {
+	if lb.randomInt == -1 {
+		lb.Predict(len(dbs))
+	}
+	randomInt := lb.randomInt
+	lb.randomInt = -1
```

Comment:
this is not concurrent safe, why do we need this number anw?

Reply:
we need it for testing

Reply:
srry forgot about concurrency

Reply:
Do we really need this? 
For Resolve function?